avm/res/network/azure-firewall/main.bicep

```diff
@@ -621,3 +621,20 @@ type networkRuleCollectionType = {
     }[]
   }
 }[]?
+
+type hubIPAddressesType = {
+  @description('Optional. List of private IP addresses associated with AzureFirewall.')
+  privateIPAddress: string?
+  @description('Optional. List of public IP addresses associated with AzureFirewall.')
+  publicIPs: {
+    @description('Optional. The list of Public IP addresses associated with azure firewall or IP addresses to be retained.')
+    addresses: [
+      {
+        @description('Optional. Public IP')
+        address: string?
+      }
+    ]?
+    @description('Optional. Public IP address count.')
+    count: int?
+  }?
+}?
```

Comment:
Please move that last `?` to the parameter. Makes it easier to udnerstand what's happening

Reply:
Unclear on what is needed here?

Reply:
The idea would be to change this

```bicep
@description('Conditional. IP addresses associated with AzureFirewall. Required if `virtualHubId` is supplied.')
param hubIPAddresses hubIPAddressesType

type hubIPAddressesType = {
  (...)
}?
```

to this

```bicep
@description('Conditional. IP addresses associated with AzureFirewall. Required if `virtualHubId` is supplied.')
param hubIPAddresses hubIPAddressesType?

type hubIPAddressesType = {
  (...)
}
```

i.e. moving the question mark. 

The perk here: When the question mark (or e.g. array brackets) are added to the parameter rather than the type, one can see directly from looking at the parameter if the parameter is nullable or not (or an array or not).

Reply:
Got it. Fixed.